core/navigation.js

```diff
@@ -22,6 +22,104 @@ goog.provide('Blockly.Navigation');
 
 Blockly.Navigation.connection = null;
 
+Blockly.Navigation.curCategory = null;
+
+/************************/
+/** Toolbox Navigation **/
+/************************/
+
+/**
+ * Set the first category as the first category in the list.
+ */
+Blockly.Navigation.focusToolbox = function() {
+  var workspace = Blockly.getMainWorkspace();
+  var toolbox = workspace.getToolbox();
+  if (workspace) {
+    Blockly.Navigation.curCategory = toolbox.tree_.firstChild_;
+  }
+  toolbox.tree_.setSelectedItem(Blockly.Navigation.curCategory);
+};
+
+/**
+ * Select the next category.
+ * Taken from basenode.js
```

Comment:
What is basenode.js?

Reply:
closure/goog/ui/tree/basenode.js. We already have keyboard navigation for the toolbox because we are using closure's tree structure. This functionality however is tied to the up, down, left, right keys and is not something we can modify so I moved the functionality to this file so we can change during testing if we want.